.github/workflows/ci-cd.yml

```
@@ -20,10 +19,10 @@ jobs:
     # Tests will use in-memory alternatives
     
     steps:
-    - uses: actions/checkout@v4
+    - uses: actions/checkout@v3
```

actionlint (1.7.7)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)